default/python/universe_generation/teams.py

```diff
@@ -0,0 +1,111 @@
+from logging import warning
+
+from common.configure_logging import redirect_logging_to_freeorion_logger
+
+# Logging is redirected before other imports so that import errors appear in log files.
+redirect_logging_to_freeorion_logger()
+
+import freeorion as fo
+
+from operator import itemgetter
+from empires import home_system_layout
+
+
+def home_system_team_core(home_systems, teams):
+    """
+    Choose core for teams
+    Returns map from team to core home system
+    """
+    print("Teams: ", teams)
+    # sort all home systems by distance
+    home_systems_distances = {}
+    for hs1 in home_systems:
+        for hs2 in home_systems:
+            if hs1 < hs2:
+                dist = fo.jump_distance(hs1, hs2)
+                home_systems_distances[(hs1, hs2)] = dist
+    home_systems_sorted = sorted(home_systems_distances.items(), key=itemgetter(1), reverse=True)
+    print("Home systems sorted: ", home_systems_sorted)
+
+    result = {}
+    if len(teams) == 0 or len(home_systems_sorted) == 0:
+        pass
+    elif len(teams) == 1:
```

Comment:
Does it mean that everyone will be in one team?  Or `teams` is just all members of one team? 

Let's move teams count checks to the top of the function.

Reply:
Yes, it means only 1 team is in the game. Other players doesn't part of any teams. I can only check empty teams case on the top of the function.